pkg/deploy/lattice/service_network_manager.go

```diff
@@ -53,209 +54,201 @@ type defaultServiceNetworkManager struct {
 		deleteServiceNetworkVpcAssociationInput := vpclattice.DeleteServiceNetworkVpcAssociationInput{
-			ServiceNetworkVpcAssociationIdentifier: service_networkAssociatedWithCurrentVPCId,
+			ServiceNetworkVpcAssociationIdentifier: serviceNetworkAssociatedWithCurrentVpcId,
 		}
-		glog.V(2).Infof("DeleteServiceNetworkVpcAssociationInput >>>> %v\n", deleteServiceNetworkVpcAssociationInput)
-		resp, err := vpcLatticeSess.DeleteServiceNetworkVpcAssociationWithContext(ctx, &deleteServiceNetworkVpcAssociationInput)
-		glog.V(2).Infof("DeleteServiceNetworkVPCAssociationResp: service_network %v , resp %v, err %v \n", snName, resp, err)
+		m.log.Debugf("Deleting ServiceNetworkVpcAssociation %s", *serviceNetworkAssociatedWithCurrentVpcId)
+		_, err := vpcLatticeSess.DeleteServiceNetworkVpcAssociationWithContext(ctx, &deleteServiceNetworkVpcAssociationInput)
 		if err != nil {
-			glog.V(2).Infof("Failed to delete association for %v, err: %v \n", snName, err)
+			m.log.Debugf("Failed to delete association for %s, err: %s", snName, err)
 		}
 		// retry later to check if VPC disassociation workflow finishes
 		return errors.New(LATTICE_RETRY)
-
 	}
 
 	// check if this VPC is the one created the service network
 	needToDelete := false
-	if service_networkSummary.Tags != nil {
-		vpcOwner, ok := service_networkSummary.Tags[latticemodel.K8SServiceNetworkOwnedByVPC]
+	if serviceNetworkSummary.Tags != nil {
+		vpcOwner, ok := serviceNetworkSummary.Tags[latticemodel.K8SServiceNetworkOwnedByVPC]
 		if ok && *vpcOwner == config.VpcID {
 			needToDelete = true
 		} else {
 			if ok {
-				glog.V(2).Infof("Skip deleting, the service network[%v] is created by VPC %v", snName, *vpcOwner)
+				m.log.Debugf("Skip deleting, service network %s is created by VPC %s", snName, *vpcOwner)
 			} else {
-				glog.V(2).Infof("Skip deleting, the service network[%v] is not created by K8S, since there is no tag", snName)
+				m.log.Debugf("Skip deleting, service network %s is not created by K8S, since there is no tag", snName)
 			}
 		}
 	}
 
 	if needToDelete {
-
 		if len(assocResp) != 0 {
-			glog.V(2).Infof("Retry deleting %v later, due to service network still has VPCs associated", snName)
+			m.log.Debugf("Retry deleting service network %s later since it still has VPCs associated", snName)
 			return errors.New(LATTICE_RETRY)
 		}
 
+		m.log.Debugf("Deleting service network %s", snName)
 		deleteInput := vpclattice.DeleteServiceNetworkInput{
-			ServiceNetworkIdentifier: &service_networkID,
+			ServiceNetworkIdentifier: &serviceNetworkId,
 		}
-		glog.V(2).Infof("DeleteServiceNetworkWithContext: service_network %v", snName)
 		resp, err := vpcLatticeSess.DeleteServiceNetworkWithContext(ctx, &deleteInput)
-		glog.V(2).Infof("DeleteServiceNetworkWithContext: service_network %v , resp %v, err %v \n", snName, resp, err)
 		if err != nil {
+			m.log.Debugf("Failed to delete service network %s due to %s", snName, resp)
 			return errors.New(LATTICE_RETRY)
```

Comment:
you can return wrapped retry error, it will trigger retry also will print error cause to log in Retry handler
```go
return fmt.Errorf("%w: failed to delete service network: %s: %s", RetryErr, snName, resp.Status)
```

Reply:
RetryErr is a real thing btw, https://github.com/aws/aws-application-networking-k8s/blob/main/pkg/deploy/lattice/error.go